pkg/abstractions/endpoint/buffer.go

```diff
@@ -176,6 +173,11 @@ func (rb *RequestBuffer) processRequests() {
 				continue
 			}
 
+			if req.ctx.Request().Context().Err() != nil {
```

Comment:
If the client hung up by the time we're handling this request (error is not nil even if the context is just cancelled), then we should just mark this task as canceled and move on to the next item in the request buffer.